subsys/bluetooth/host/mesh/cfg_srv.c

```diff
@@ -3057,20 +3057,20 @@ static void hb_publish(struct k_work *work)
 		return;
 	}
 
-	hb_send(model);
-
 	if (cfg->hb_pub.count == 0) {
 		return;
 	}
 
-	if (cfg->hb_pub.count != 0xffff) {
-		cfg->hb_pub.count--;
-	}
-
 	period_ms = hb_pwr2(cfg->hb_pub.period, 1) * 1000;
 	if (period_ms) {
```

Comment:
One thing I'd consider adding (as a separate patch) is && cfg->hb_pub.count > 1 to this condition, so that we don't schedule the delayed work if this was the last heartbeat message.

Reply:
Actually, > 1 is not sufficient since you need to deal with 0xffff as well (which means "send heartbeat messages indefinitely")

Reply:
So actually extending the condition with "&& cfg->hb_pub.count != 1" would likely suffice

Reply:
good catch!